test/core/resource_quota/arena_test.cc

```diff
@@ -271,6 +271,24 @@ TEST(ArenaTest, ConcurrentMakePooled) {
   }
 }
 
+struct Foo {
+  explicit Foo(int x) : p(std::make_unique<int>(x)) {}
+  std::unique_ptr<int> p;
+};
+
+template <>
+struct ArenaContextType<Foo> {
+  static void Destroy(Foo* p) { p->~Foo(); }
```

Comment:
I would assume that this is the default destruction behavior we'd want for most contexts.  Is there some way to make this the default, so that individual types don't need to specify it?

Reply:
You'd be surprised, and that default would be very crashy.

Also, I think the explicitness is good here: ways of making it optional will miss typos and compile just fine.